src/services/upper-bar-manager/move-controls-manager.ts

```diff
@@ -0,0 +1,58 @@
+import { KalturaPlayer, Logger, ui } from '@playkit-js/kaltura-player-js';
+import { UpperBarManager } from './upper-bar-manager';
+import { IconDto } from './models/icon-dto';
+
+const { redux } = ui;
+
+export class MoveControlsManager {
+  private readonly player: KalturaPlayer;
+  private readonly logger: Logger;
+  private store: any;
+  private upperBarManager: UpperBarManager;
+  private currentState: any;
+  private iconIds: Map<string, number>;
+
+  constructor(player: KalturaPlayer, logger: Logger, upperBarManager: UpperBarManager) {
+    this.player = player;
+    this.logger = logger;
+    this.store = redux.useStore();
+    this.upperBarManager = upperBarManager;
+    this.store.subscribe(this.handleStoreChange.bind(this));
+    this.currentState = this.store.getState();
+    this.iconIds = new Map();
+  }
+
+  private get bottomBarRegistryManager(): any {
+    return (this.player.getService('bottomBarRegistryManager') as any) || undefined;
+  }
+
+  private get state(): any {
+    return this.store.getState();
+  }
+
+  private handleStoreChange(): void {
+    const newState = this.state;
+    const bottomBarRegistryManager = this.bottomBarRegistryManager;
+    if (bottomBarRegistryManager && this.currentState.bottomBar !== newState.bottomBar) {
```

Comment:
why is the bottomBar reference different between states ?

Reply:
you mean it's better to compare the `controlsToMove` arrays instead of the entire `bottomBar` object?

Reply:
no, i mean you're comparing references and i want to understand why :)

Reply:
i don't want/need to do the logic inside the `if` block for **any** changes in the store, it would be an overkill. I want to execute this logic only if there was a change in the `bottomBar` state.